### PR TITLE
Clean up null handling in UnitTestingRemoteHostClientWrapper

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs [hunk 1]
```diff
@@ -6,42 +6,67 @@
...
 
+        [MemberNotNullWhen(false, nameof(UnderlyingObject))]
         public bool IsDefault => UnderlyingObject == null;
 
         public static async Task<UnitTestingRemoteHostClientWrapper?> TryGetClientAsync(HostWorkspaceServices services, CancellationToken cancellationToken = default)
-            => new UnitTestingRemoteHostClientWrapper(await RemoteHostClient.TryGetClientAsync(services, cancellationToken).ConfigureAwait(false));
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(services, cancellationToken).ConfigureAwait(false);
+            if (client is null)
+                return null;
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs:32]: 📝 This is the only behavior change. Previously it returned a non-null object with `IsDefault` set, and now it returns `null`.

Comment (Cosifne) [reply src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs:32]: @shyamnamboodiripad 
I remember in SBD/LUT it is checking the 'IsDefault' property,  we might need to change these checks to checking null after this PR is checking in?

Comment (shyamnamboodiripad) [reply src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs:32]: @Cosifne Looking at the references for [`TryGetClientAsync`](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=cn1hrhmmy4q1&file=ExternalAccess%5cUnitTesting%5cApi%5cUnitTestingRemoteHostClientWrapper.cs), all call sites are checking for both `HasValue` and `IsDefault`. So I think this should be ok.

Comment (sharwell) [reply src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs:32]: @shyamnamboodiripad that's exactly what I was hoping to hear!

**Pull request merged**